### DISCUSSION TITLE
Thank-you!

### DISCUSSION THREAD

**Discussion by ahbarnett** [JuliaMath/Bessels.jl#73] [General]
Dear Michael,
Just to say thank-you for writing a high-quality native-Julia clean implementation of these essential functions. K_0 was very useful in a little library for a new collaboration,
https://github.com/ahbarnett/YukBIE2D.jl
and I'll continue to depend on your package.
My timing tests were very roughly:
SpecialFunctions.jl  - 280 ns/eval
BesselK.jl  - 200 ns/eval and gave Nan for arg>8.5
Bessels.jl  - 15 ns/eval
Thus on 8 cores I got 0.5 Geval/s, making me a happy user.
I also like how clear (and non-abstracted) your code was, as an experienced programmer but new to Julia. The many layers of abstraction in most .jl packages make them almost impossible to understand or modify, sadly.
Best wishes,  Alex

**Comment by heltonmc** [comment]
Thank you for the kind words! Very appreciated 🙏 . And of course this wouldn't have been able to be done without Oscar's contributions and also from Chris (owner of BesselK.jl) for contributing some of his code for besselk.

Let me know if there are any other functions you would like to see! We just merged I_0 for complex arguments and working on K_0 for complex arguments now. Hopefully, that will be merged in the next couple of weeks!

Best,
Michael
